catalog/imaging/imaging-postprocess-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,574 @@
+
+          if (storeOriginalImage) {
+            ProcessResource originalImageProcessResource =
+                createOriginalImage(
+                    renderImageUsingOriginalDataModel(byteSource.openStream()), metacard);
+
+            createdItems.add(new ProcessCreateItemImpl(originalImageProcessResource, metacard));
+          }
+        }
+      } catch (IOException | NitfFormatException | RuntimeException e) {
+        LOGGER.debug("An error occured when rendering a nitf for {}", processResource.getName(), e);
+      } catch (InterruptedException e) {
+        LOGGER.error("Rendering failed for {}", processResource.getName(), e);
+        Thread.currentThread().interrupt();
+        throw new RuntimeException(e);
+      }
+    }
+    return createdItems.stream();
+  }
+
+  private void handleProcessUpdateItem(List<ProcessUpdateItem> processUpdateItems) {
+    List<ProcessUpdateItem> updateItems =
+        processUpdateItems
+            .stream()
+            .flatMap(this::handleProcessUpdateItem)
+            .collect(Collectors.toList());
+    processUpdateItems.addAll(updateItems);
+  }
+
+  private Stream<ProcessUpdateItem> handleProcessUpdateItem(ProcessUpdateItem processUpdateItem) {
+    List<ProcessUpdateItem> updatedItems = new ArrayList<>();
+    Metacard metacard = processUpdateItem.getMetacard();
+    Metacard originalMetacard = processUpdateItem.getOldMetacard();
+    ProcessResource processResource = processUpdateItem.getProcessResource();
+
+    if (shouldProcess(processResource)) {
+      try (TemporaryFileBackedOutputStream fbos =
+          new TemporaryFileBackedOutputStream(DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD)) {
+
+        fbos.write(IOUtils.toByteArray(processResource.getInputStream()));
+        ByteSource byteSource = fbos.asByteSource();
+        BufferedImage renderedImage = renderImage(byteSource.openStream());
+
+        if (renderedImage != null) {
+          addThumbnailToMetacard(metacard, renderedImage);
+          processUpdateItem.markMetacardAsModified();
+
+          if (createOverview) {
+            ProcessResource overviewProcessResource =
+                createOverviewResource(renderedImage, metacard);
+            updatedItems.add(
+                new ProcessUpdateItemImpl(overviewProcessResource, metacard, originalMetacard));
+          }
+
+          if (storeOriginalImage) {
+            ProcessResource originalImageProcessResource =
+                createOriginalImage(
+                    renderImageUsingOriginalDataModel(byteSource.openStream()), metacard);
+
+            updatedItems.add(
+                new ProcessUpdateItemImpl(
+                    originalImageProcessResource, metacard, originalMetacard));
+          }
+        }
+      } catch (IOException | NitfFormatException | RuntimeException e) {
+        LOGGER.debug(e.getMessage(), e);
+      } catch (InterruptedException e) {
+        LOGGER.error("Rendering failed for {}", processResource.getName(), e);
```

Comment:
❗️ Need to properly handle `InterruptedException`. See https://github.com/codice/alliance/pull/559/files#r193193376.

Reply:
Fixed